pkg/infrastructure/clusterapi/clusterapi.go

```diff
@@ -108,7 +111,9 @@ func (i *InfraProvider) Provision(ctx context.Context, dir string, parents asset
 			WorkersAsset:     workersAsset,
 		}
 
-		if err := p.PreProvision(ctx, preProvisionInput); err != nil {
+		preCtx, cancel := context.WithTimeout(ctx, timeout)
+		defer cancel()
+		if err := p.PreProvision(preCtx, preProvisionInput); err != nil {
```

Comment:
I'm going to remove these timeouts from the hooks and only keep the timeout caps on the provisioning with the capi system. We don't need to handle hook failures with timeouts, instead we can leave that up to the hook implementors. For CAPI provisioning on the other hand, we need a timeout to prevent endlessly spinning.